File: components/application-mgt/org.wso2.carbon.device.application.mgt.core/src/main/java/org/wso2/carbon/device/application/mgt/core/config/Configuration.java
/*
 *   Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 *   WSO2 Inc. licenses this file to you under the Apache License,
 *   Version 2.0 (the "License"); you may not use this file except
 *   in compliance with the License.
 *   You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 *   Unless required by applicable law or agreed to in writing,
 *   software distributed under the License is distributed on an
 *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 *   KIND, either express or implied.  See the License for the
 *   specific language governing permissions and limitations
 *   under the License.
 *
 */
package org.wso2.carbon.device.application.mgt.core.config;

import org.wso2.carbon.device.application.mgt.common.config.MDMConfig;
import org.wso2.carbon.device.application.mgt.common.config.RatingConfiguration;
import org.wso2.carbon.device.application.mgt.common.config.UIConfiguration;
import org.wso2.carbon.device.application.mgt.common.config.LifecycleState;

import java.util.List;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlElementWrapper;
import javax.xml.bind.annotation.XmlRootElement;

/**
 * Represents the Application Management Configuration.
 */
@XmlRootElement(name = "ApplicationManagementConfiguration")
public class Configuration {

    private String datasourceName;

    private List<Extension> extensions;

    private List<LifecycleState> lifecycleStates;

    private UIConfiguration uiConfiguration;

    private List<String> appCategories;

    private RatingConfiguration ratingConfiguration;

    private MDMConfig mdmConfig;

    @XmlElement(name = "DatasourceName", required = true)
    public String getDatasourceName() {
        return datasourceName;
    }

    public void setDatasourceName(String datasourceName) {
        this.datasourceName = datasourceName;
    }

    @XmlElementWrapper(name = "Extensions")
    @XmlElement(name = "Extension")
    public List<Extension> getExtensions() {
        return extensions;
    }

    public void setExtensions(List<Extension> extensions) {
        this.extensions = extensions;
    }

    @XmlElementWrapper(name = "LifecycleStates")
    @XmlElement(name = "LifecycleState")
    public List<LifecycleState> getLifecycleStates() {
        return lifecycleStates;
    }

    public void setLifecycleStates(List<LifecycleState> lifecycleStates) {
        this.lifecycleStates = lifecycleStates;
    }

    @XmlElement(name = "UIConfigs")
    public UIConfiguration getUiConfiguration() {
        return uiConfiguration;
    }

    public void setUiConfiguration(UIConfiguration uiConfiguration) {
        this.uiConfiguration = uiConfiguration;
    }

    @XmlElement(name = "RatingConfig")
    public RatingConfiguration getRatingConfiguration() { return ratingConfiguration; }

    public void setRatingConfiguration(
            RatingConfiguration ratingConfiguration) { this.ratingConfiguration = ratingConfiguration; }

    @XmlElementWrapper(name = "AppCategories")
    @XmlElement(name = "Category")
    public List<String> getAppCategories() {
        return appCategories;
    }

    public void setAppCategories(List<String> appCategories) {
        this.appCategories = appCategories;
    }

    @XmlElement(name = "MDMConfig", required = true)
    public MDMConfig getMdmConfig() { return mdmConfig; }

    public void setMdmConfig(MDMConfig mdmConfig) {
        this.mdmConfig = mdmConfig;
    }
}

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.platform.configuration/configuration.js
/*
 * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * WSO2 Inc. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 *
 *
 * Copyright (c) 2019, Entgra (pvt) Ltd. (http://entgra.io) All Rights Reserved.
 *
 * Entgra (pvt) Ltd. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

function onRequest(context) {
    var log = new Log("configuration.js");
    var utility = require("/app/modules/utility.js").utility;
    var deviceModule = require("/app/modules/business-controllers/device.js")["deviceModule"];
    var devicemgtProps = require("/app/modules/conf-reader/main.js")["conf"];

    //get all device types
    var isAuthorized = false;
    if (userModule.isAuthorized("/permission/admin/device-mgt/notifications/view")) {
        isAuthorized = true;
    }
    var deviceTypesArray = [];
    var typesListResponse = deviceModule.getDeviceTypes();
    if (typesListResponse["status"] == "success") {
        var data = typesListResponse["content"];
        if (data) {
            for (var i = 0; i < data.length; i++) {
                var deviceTypeName = data[i].name;
                var deviceTypeLabel = deviceTypeName.charAt(0).toUpperCase() + deviceTypeName.slice(1);
                var configUnitName = utility.getTenantedDeviceUnitName(deviceTypeName, "platform.configuration");
                if (configUnitName) {
                    var deviceTypeConfig = utility.getDeviceTypeConfig(deviceTypeName);
                    if (deviceTypeConfig) {
                        deviceTypeLabel = deviceTypeConfig.deviceType.label;
                    }
                }
                deviceTypesArray.push({
                    name: deviceTypeName,
                    label: deviceTypeLabel,
                    unitName: configUnitName
                });
            }
        }
    }
    var geoServicesEnabled = devicemgtProps.serverConfig.geoLocationConfiguration.enabled;
    return {
        "geoServicesEnabled": geoServicesEnabled,
        "deviceTypes": deviceTypesArray,
        "isAuthorized": isAuthorized
    };
}
